src/main/java/org/pac4j/dropwizard/Pac4jFactory.java

```diff
@@ -329,44 +363,84 @@ public void setSessionEnabled(boolean sessionEnabled) {
     }
 
     public Config build() {
-        Clients client = new Clients();
-        JaxRsConfig config = new JaxRsConfig();
+        // either the whole Config is built from a ConfigFactory class, or we initialize a default JaxRsConfig
+        final JaxRsConfig config;
+        if (configClass != null && configProperties != null) {
+            config = (JaxRsConfig) ConfigBuilder.build(configClass, configProperties);
+        } else {
+            config = new JaxRsConfig();
+        }
+        Clients newClients = config.getClients();
+        if (newClients == null) {
+            newClients = new Clients();
+            config.setClients(newClients);
+        }
+        if (newClients.getClients() == null) {
+            newClients.setClients(new ArrayList<>());
+        }
+        config.setClients(newClients);
 
-        config.setClients(client);
         config.setDefaultClients(defaultClients);
 
+        // we can take the clients built from the properties
+        if (clientsProperties != null && clientsProperties.size() > 0) {
+            final PropertiesConfigFactory propertiesConfigFactory = new PropertiesConfigFactory(clientsProperties);
+            final Config propertiesConfig = propertiesConfigFactory.build();
+            config.getClients().getClients().addAll(propertiesConfig.getClients().getClients());
+        }
+        // and the clients directly defined in the YAML file
+        if (clients != null) {
+            config.getClients().getClients().addAll(clients);
+        }
+
         if (callbackUrl != null) {
-            client.setCallbackUrl(callbackUrl);
+            newClients.setCallbackUrl(callbackUrl);
         }
         if (clientNameParameter != null) {
-            client.setClientNameParameter(clientNameParameter);
+            newClients.setClientNameParameter(clientNameParameter);
+        }
+        newClients.setUrlResolver(urlResolver);
+        if (authorizationGenerators != null) {
+            newClients.setAuthorizationGenerators(authorizationGenerators);
```

Comment:
Does this overwrite the whole collection or only add the arguments to the existing collection of `AuthorizationGenerators`? The latter is expected I guess